site/_posts/2020-05-06-using-java-to-manage-fusionauth.md

```diff
@@ -0,0 +1,743 @@
+import com.inversoft.error.Errors;
+import io.fusionauth.client.FusionAuthClient;
+import io.fusionauth.domain.api.UserResponse;
+import com.inversoft.rest.ClientResponse;
+
+public class RetrieveUserInfo {
+
+    public static void main(String[] args) {
+
+        ApplicationProperties.setupProperties();
+
+        // Initiating the client
+        FusionAuthClient client = new FusionAuthClient(ApplicationProperties.getApiKey(),
+                ApplicationProperties.getFusionAuthURL());
+
+        UUID userId = UUID.fromString("5eb46e47-927c-41ac-ae60-ed88e3840edd");
+
+        // Using the returned ClientResponse object
+        ClientResponse<UserResponse, Errors> response = client.retrieveUser(userId);
+
+        if (response.wasSuccessful()) {
+            // Outputting the user's profile details
+            System.out.println(response.successResponse.user);
+
+        } else {
+            // Handling errors
+            System.out.println(response.errorResponse);
+
+        }
+    }
+}
+```
+
+If you run the above code, here's the result we get on the console:
+
+```json
+{
+  "encryptionScheme" : null,
+  "factor" : null,
+  "id" : "812478d4-c99d-4e55-9606-9e58e83938d3",
+  "password" : null,
+  "passwordChangeRequired" : false,
+  "passwordLastUpdateInstant" : 1587287741465,
+  "salt" : null,
+  "verified" : true,
+  "preferredLanguages" : [ ],
+  "memberships" : [ ],
+  "registrations" : [ {
+    "data" : { },
+    "preferredLanguages" : [ ],
+    "tokens" : { },
+    "applicationId" : "3c219e58-ed0e-4b18-ad48-f4f92793ae32",
+    "authenticationToken" : null,
+    "cleanSpeakId" : null,
+    "id" : "7c105c78-c49d-40c1-a5a1-fcff013fd4eb",
+    "insertInstant" : 1587299377006,
+    "lastLoginInstant" : null,
+    "roles" : [ "admin" ],
+    "timezone" : null,
+    "username" : null,
+    "usernameStatus" : "ACTIVE",
+    "verified" : true
+  } ],
+  "active" : true,
+  "birthDate" : null,
+  "cleanSpeakId" : null,
+  "data" : { },
+  "email" : "fusionjava@gmail.com",
+  "expiry" : null,
+  "firstName" : null,
+  "fullName" : null,
+  "imageUrl" : null,
+  "insertInstant" : 1587287741325,
+  "lastLoginInstant" : null,
+  "lastName" : null,
+  "middleName" : null,
+  "mobilePhone" : null,
+  "parentEmail" : null,
+  "tenantId" : "31626538-3731-3762-3664-303931343339",
+  "timezone" : null,
+  "twoFactorDelivery" : "None",
+  "twoFactorEnabled" : false,
+  "twoFactorSecret" : null,
+  "username" : "fusionjava",
+  "usernameStatus" : "ACTIVE"
+}
+```
+This would be useful for displaying on their profile page in the todo application, for example.
+
+
+## Log in a user
+
+Let’s say we want to use the FusionAuth Java client to login the user we’d created. We'd do this if we wanted to log a user in using our own application, rather than using [FusionAuth's OAuth capabilities](https://fusionauth.io/docs/v1/tech/oauth/) and the Authorization Code grant. You might do this if you were building a command line tool to add todos.
```

Comment:
Log in is a verb in the in the first sentence and should be two words. Log in.
```suggestion
Let’s say we want to use the FusionAuth Java client to log in the user we’d created. We'd do this if we wanted to log a user in using our application, rather than using [FusionAuth's OAuth capabilities](https://fusionauth.io/docs/v1/tech/oauth/) and the Authorization Code grant. You might do this if you were building a command line tool to add todos.
```

Reply:
good catch